src/map/atcommand.c

```diff
@@ -5380,25 +5379,31 @@ ACMD(dropall)
 		}
 	}
 
+	int count = 0, count_skipped = 0;
 	for (int i = 0; i < sd->status.inventorySize; i++) {
-		if (sd->status.inventory[i].amount) {
+		if (sd->status.inventory[i].amount > 0) {
 			struct item_data *item_data = itemdb->exists(sd->status.inventory[i].nameid);
 			if (item_data == NULL) {
 				ShowWarning("Non-existant item %d on dropall list (account_id: %d, char_id: %d)\n", sd->status.inventory[i].nameid, sd->status.account_id, sd->status.char_id);
 				continue;
 			}
+			
 			if (!pc->candrop(sd, &sd->status.inventory[i]))
 				continue;
+			
 			if (type == -1 || type == item_data->type) {
 				if (sd->status.inventory[i].equip != 0)
 					pc->unequipitem(sd, i, PCUNEQUIPITEM_RECALC | PCUNEQUIPITEM_FORCE);
-				count += sd->status.inventory[i].amount;
-				pc->dropitem(sd, i, sd->status.inventory[i].amount);
+				
+				if (pc->dropitem(sd, i, sd->status.inventory[i].amount) != 0)
+					count += sd->status.inventory[i].amount;
```

Comment:
sd->status.inventory.amount should be saved in variable before and then added here.
Using it after dropping item can result in crash/garbage value.
See @Lemongrass3110 comment for more details

Reply:
no crash because here no pointer, but look like it will be 0

Reply:
fixed.